ios/FluentUI/Navigation/NavigationBar.swift

```diff
@@ -47,57 +47,85 @@ open class NavigationBarTopSearchBarAttributes: NavigationBarTopAccessoryViewAtt
     }
 }
 
+// MARK: - NavigationBarTitleAccessory
+
+@objc(MSFNavigationBarTitleAccessoryDelegate)
+/// Handles user interactions with a `NavigationBar` with an accessory.
+public protocol NavigationBarTitleAccessoryDelegate {
+    @objc func navigationBarDidTapOnTitle(_ sender: NavigationBar)
+}
+
+/// The specifications for an accessory to show in the title or subtitle of the navigation bar.
+@objc(MSFNavigationBarTitleAccessory)
+open class NavigationBarTitleAccessory: NSObject {
+    /// Specifies a location where the title accessory should appear within the navigation bar.
+    @objc(MSFNavigationBarTitleAccessoryLocation)
+    public enum Location: Int {
+        case title
+        case subtitle
+    }
+
+    /// The style of title accessory to show.
+    @objc(MSFNavigationBarTitleAccessoryStyle)
+    public enum Style: Int {
+        case disclosure
+        case downArrow
+    }
+
+    /// The location of the accessory.
+    public let location: Location
+    /// The style of the accessory.
+    public let style: Style
+    /// A delegate that handles title press actions.
+    public weak var delegate: NavigationBarTitleAccessoryDelegate?
+
+    public init(location: Location, style: Style, delegate: NavigationBarTitleAccessoryDelegate? = nil) {
+        self.location = location
+        self.style = style
+        self.delegate = delegate
+    }
+}
+
+// MARK: - NavigationBarBackButtonDelegate
+/// Handles presses from the back button shown with a leading-aligned title.
+@objc(MSFNavigationBarBackButtonDelegate)
+protocol NavigationBarBackButtonDelegate {
+    func backButtonWasPressed()
+}
+
 // MARK: - NavigationBar
 
 /// UINavigationBar subclass, with a content view that contains various custom UIElements
 /// Contains the MSNavigationTitleView class and handles passing animatable progress through
 /// Custom UI can be hidden if desired
 @objc(MSFNavigationBar)
-open class NavigationBar: UINavigationBar, TokenizedControlInternal {
+open class NavigationBar: UINavigationBar, TokenizedControlInternal, TwoLineTitleViewDelegate {
     /// If the style is `.custom`, UINavigationItem's `navigationBarColor` is used for all the subviews' backgroundColor
     @objc(MSFNavigationBarStyle)
     public enum Style: Int {
```

Comment:
The style should probably live in `NavigationBarTokenSet` in an extension of `NavigationBar` the goal being to group all styling and design values in a single file making it easier for us and partners to find them.

Reply:
As I explained in [this comment](https://github.com/microsoft/fluentui-apple/pull/1684#discussion_r1154974257), `NavigationBar.Style` isn't directly related to tokens. If some day we end up overhauling the idea of *TokenSet.swift files, the idea of having to put `NavigationBar.Style` back into the replacement feels very wrong.